programming-extensions/programming-extension-dataspaces/src/main/java/org/objectweb/proactive/extensions/dataspaces/vfs/VFSNodeScratchSpaceImpl.java

```diff
@@ -117,17 +118,21 @@ private AppScratchSpaceImpl(final long appId) throws FileSystemException {
 
         public void close() throws FileSystemException {
             logger.debug("Closing application scratch space");
+
             try {
+                final int filesNumber = spaceFile.delete(Selectors.SELECT_ALL);
+                logger.debug("Deleted " + filesNumber + " files in scratch application directory");
+            } catch (org.apache.commons.vfs2.FileSystemException e) {
+                logger.warn("Could not delete " + spaceFile, e);
```

Comment:
> Can we resolve that error additionally to logging it?

I agree that catching such an exception and logging it only is not nice but it's a tradeoff with the fact that if I don't catch the exception at this level, it will have to be done at the Scheduler/node level. Unfortunately, at higher level nothing can be done when deletion fails and same code will be duplicated. Also, it is important to notice that the class `VFSNodeScratchSpaceImpl` is used in the context of dataspaces only. Besides, it depicts a temporary storage folder (called scratch space).

> Is it important to delete files?

I know it is important but what do you want to do if the deletion operation fails because the file is read-only or a process hold a lock on it and you have not changed the permissions by yourself and you don't have the hand over this last process?

> Can conflicts happen if we keep files?

No, it should not. Folders are created per (App ID, ProActive node runtime ID, JVM ID).